ccvm_simulators/solvers/ccvm_solver.py

```diff
@@ -338,3 +348,150 @@ def machine_energy(self, machine: str, machine_parameters: dict = None):
             )
 
         return energy_method(machine_parameters)
+
+    ##############################
+    ### MACHINE TIME FUNCTIONS ###
+    ##############################
+
+    def _validate_machine_time_dataframe_columns(self, dataframe):
+        """Validates that the given dataframe contains the required columns when
+        calculating optics machine time on DL-CCVM and MF-CCVM solvers.
+
+        Args:
+            dataframe (DataFrame): The dataframe to validate.
+
+        Raises:
+            ValueError: If the given dataframe is missing any of the required columns.
+        """
+        required_columns = ["pp_time", "iterations"]
+
+        missing_columns = [
+            col for col in required_columns if col not in dataframe.columns
+        ]
+
+        if missing_columns:
+            raise ValueError(
+                f"The given dataframe is missing the following columns: {missing_columns}"
+            )
+
+    def _cpu_machine_time(self, **_):
```

Comment:
Good observation! You are right that these functions are the same and that the repetition is not ideal. I will modify this as you suggested. One thing I might do differently is to name it `_conventional_hardware_machine_time` or `_cpu_gpu_machine_time`, to focus on what it is rather than what it isn't.
One thing is that in the future these two may diverge. Since we don't know that to be the case, I still think your idea is good, and I'll choose the `_cpu_gpu_machine_time` name to make it clear that this is where we would split the functions if they diverge.